src/container/streamAnalyticsManager.ts

```diff
@@ -4,40 +4,129 @@
 import StreamAnalyticsManagementClient = require("azure-arm-streamanalytics");
 import { StreamingJob } from "azure-arm-streamanalytics/lib/models";
 import { StreamingJobs } from "azure-arm-streamanalytics/lib/operations";
+import * as fse from "fs-extra";
 import * as request from "request-promise";
 import * as vscode from "vscode";
 import { Constants } from "../common/constants";
 import { UserCancelledError } from "../common/UserCancelledError";
 import { Utility } from "../common/utility";
-import { AzureAccount } from "../typings/azure-account.api";
+import { AzureAccount, AzureSession, AzureSubscription } from "../typings/azure-account.api";
 import { StreamAnalyticsPickItem } from "./models/streamAnalyticsPickItem";
 
+export enum ASAUpdateStatus {
+    Idle,
+    CheckingUpdate,
+    Updating,
+}
+
 export class StreamAnalyticsManager {
+    public static getInstance(): StreamAnalyticsManager {
+        if (!StreamAnalyticsManager.instance) {
+            StreamAnalyticsManager.instance = new StreamAnalyticsManager();
+            return StreamAnalyticsManager.instance;
```

Comment:
actually this line could be removed right? just return outside.

Reply:
OK